targets/TARGET_Freescale/TARGET_KLXX/TARGET_KL46Z/device/TOOLCHAIN_ARM_STD/MKL46Z4.sct

```diff
@@ -8,24 +24,30 @@
 #  endif
 #endif
 
-#define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE              (MBED_CONF_TARGET_BOOT_STACK_SIZE + VECTOR_SIZE)
+
+; 8_byte_aligned(48 vect * 4 bytes) =  8_byte_aligned(0xC0) = 0xC0
+#define VECTOR_SIZE                   0xC0
 
-LR_IROM1 0x00000000 0x40000  {    ; load region size_region (256k)
-  ER_IROM1 0x00000000 0x40000  {  ; load address = execution address
+#define MBED_RAM1_START (MBED_RAM_START + VECTOR_SIZE)
+#define MBED_RAM1_SIZE  (MBED_RAM_SIZE - VECTOR_SIZE)
+
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region (256k)
+  ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address
    *.o (RESET, +First)
    *(InRoot$$Sections)
    .ANY (+RO)
   }
-  ; 8_byte_aligned(48 vect * 4 bytes) =  8_byte_aligned(0xC0) = 0xC0
+
   ; 0x8000 - 0xC0 = 0x7F40
-  RW_IRAM1 0x1FFFE0C0 0x7F40  {
+  RW_IRAM1 MBED_RAM1_START MBED_RAM1_SIZE  {
    .ANY (+RW +ZI)
   }
-  
-  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (0x1FFFE000+0x8000-Stack_Size-AlignExpr(ImageLimit(RW_IRAM1), 16)) { ; Heap region growing up
+
+  ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (MBED_RAM_START + MBED_RAM_SIZE - MBED_CONF_TARGET_BOOT_STACK_SIZE - AlignExpr(ImageLimit(RW_IRAM1), 16)) { ; Heap region growing up
```

Comment:
I do not think this is correct. One way of calculating the heap size is:
`MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM1_START)`